File: James/221222_study/프로_42862_체육복.java
package Day03;

import java.util.*;
import java.util.stream.Collectors;

public class 프로_42862_체육복 {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;

        Set<Integer> newLost, newReserve, set;

        newLost = Arrays.stream(lost).boxed().collect(Collectors.toSet());
        newReserve = Arrays.stream(reserve).boxed().collect(Collectors.toSet());

        set = new HashSet<>(newReserve);
        set.retainAll(newLost);

        newLost.removeAll(set); 
        newReserve.removeAll(set);

        List<Integer> newLost2 = new ArrayList<>(newLost);
        List<Integer> newReserve2 = new ArrayList<>(newReserve);

        for(int i : newReserve2) {
            if(newLost2.contains(i - 1))
                newLost2.remove(i - 1);

            else if (newLost2.contains(i + 1)) {
                newLost2.remove(i + 1);
            }
        }

        answer = n - newLost2.size();

        return answer;
    }
}

File: sammy/221222_study/프로_42862_체육복.py
# 전체 학생 수 n
# 체육복을 도난당한 학생들의 번호가 담긴 배열 lost
# 여벌의 체육복을 가져온 학생들의 번호가 담긴 배열 reverse

def solution(n, lost, reserve):
    # set으로 중복값 처리
    answer=0
    new_reserve = list(set(reserve) - set(lost))
    new_lost = list(set(lost) - set(reserve))
    
    # reserve의 +-1 값 check_lost와 비교.
    for check_reserve in new_reserve: 
       
        front = check_reserve - 1
        back = check_reserve + 1

        if front in new_lost:
            new_lost.remove(front)
            
        elif back in new_lost:
            new_lost.remove(back)

    # n에서 new_lost 뺀 값이 answer        
    answer=n - len(new_lost)

    return answer
    

# print(solution(5,[2,4],[1, 3, 5]))
# print(solution(5, [2, 4], [3]))